### ISSUE TITLE
SVI estimates are off with numpyro==0.5.0

### ISSUE THREAD

**Issue by joehoover** [pyro-ppl/numpyro#974]
I recently relocated a project to a different system and noticed some strange estimates from a relatively simple model. After a fair amount of trial and error, I traced the issue to my either my `numpyro` or `jax` versions.

On the original system, I was running `numpyro=0.4.1` and `jax=0.2.3` and on the new system I was running `numpyro=0.5.0` and `jax=0.2.8`. When I downgraded on the new system, the issue was resolved.

My model, estimation code, and data can be downloaded here [Archive.zip](https://github.com/pyro-ppl/numpyro/files/6205659/Archive.zip).

**tldr:** My model estimates K independent beta posteriors with varying numbers of observations. It's implemented with masking and plates. Before downgrading, SVI yielded substantially different posterior estimates for identical inputs. After downgrading, this issue was resolved.

For example, before downgrading, the `alpha` shape parameters estimated with SVI for a matrix with identical columns were: 

```
DeviceArray([[0.22837089, 0.32834676, 0.22467493, 0.2639365 ]], dtype=float32)
```

However, SVI run after downgrading or on my initial machine _and_ MCMC with any stack yielded:

```
DeviceArray([[0.22837183, 0.22752498, 0.22467493, 0.22676043]]
```

I do not know if this issue was caused by an API change or a bug and I also don't know if it's already been resolved. However, since it was pretty annoying to resolve, I thought I'd open an issue.

**A bit more detail**
I'm estimating a simple model expects an N x K matrix, `rho_hat`. The number of observations for each K is variable, so there is missing data in some of the columns. Using plates, I'm estimating a beta posterior for each column (model code below). In other words, I'm just estimating K independent beta posteriors in parallel.

On my initial machine, MCMC and SVI estimates were comparable. However, on my new machine, before downgrading `jax` and `numpyro`, estimates seemed to change as I increased K. 

Specifically, on the new machine, using an N x 4 input matrix where each column is identical, SVI yields substantially different estimates for the columns. This only happens with `numpyro=0.5.0` and `jax=0.2.8`. When I downgraded, the issue disappeared. 

**Model**
```python
def model(N = None, JK = None, rho_hat = None):
    
    # Initialize a plate for each 0:J group
    with numpyro.plate("J", JK):

        # Set priors on a and b within each group
                
        alpha_a = .25
        beta_a = 5
        gamma_a = numpyro.sample('gamma_a', dist.Gamma(alpha_a, beta_a))
        
        alpha_b = 8
        beta_b = 4
        gamma_b =  numpyro.sample('gamma_b', dist.Gamma(8,4))
        

        # Calculate maximum group size
        N_max = N.max().item()
        # Create a range of 0:I
        i = torch.arange(N_max).unsqueeze(-1).numpy()
        
        # Initialize a plate for each observation
        with numpyro.plate('data', N_max):
            # Mask observations that exceed sample size in the respective plate
            with numpyro.handlers.mask(mask = i < N): 
                # Estimate rho_hat
                rho_hat = numpyro.sample('rho_hat', dist.Beta(gamma_a, gamma_b), obs=rho_hat)
```

```python
def estimate_model(model, N, JK, rho_hat, method = 'SVI',
                   seed = None, num_warmup = 500, num_samples = 2000, n_iter = 10000, num_chains = 3):
    
    
    if seed is not None:
        rng_key = jax.random.PRNGKey(seed)
    else:
        rng_key = jax.random.PRNGKey(random.randint(0,10000))
    
    rng_key, rng_key_ = jax.random.split(rng_key)

    if method == 'MCMC':
    
        kernel = NUTS(model)
        mcmc = MCMC(kernel, num_warmup, num_samples, num_chains = num_chains)

        mcmc.run(rng_key_, N=N, JK = JK, rho_hat = rho_hat)

        samples = mcmc.get_samples()

        return(mcmc, samples)
    
    elif method == 'SVI':

        guide = autoguide.AutoLaplaceApproximation(model)

        svi = SVI(
            model=model,
            guide=guide,
            optim=numpyro.optim.Adam(0.0005),
            loss=Trace_ELBO(), N=N, JK=JK, rho_hat=rho_hat
        )

        init_state = svi.init(rng_key_)
        state, loss = lax.scan(lambda x, i: svi.update(x), init=init_state, xs=jnp.zeros(n_iter))
        params = svi.get_params(state)

        samples = guide.sample_posterior(rng_key, params, sample_shape = (1000,1))

        return(guide, samples)
        
    else:
        raise ValueError('`method` must be set to either "SVI" or "MCMC".')
```

**Comment by fehiepsi** [comment]
Hi @joehoover, I took a look and it seems that the loss does not convert yet. Could you try to change the number of iterations to `20000` for example?

**Comment by joehoover** [comment]
Thanks @fehiepsi! I should have tried that, but for some reason it didn't occur to me. 

Increasing the number of iterations to 20k didn't help, but the discrepancy disappears with 50k iterations.

**Comment by fehiepsi** [comment]
I think you can also increase the learning rate. :) Probably this is a random seed issue, or jax optimizers change their default values.
